### PR TITLE
AxisTensors: generic local coordinate vectors

### PR DESCRIPTION
**Pull request by simonbyrne**
Make the `Covariant`/`Contravariant` vector/tensor stuff a bit clearer.

### CODE REVIEW

**Comment by simonbyrne** [comment]
bors try

**Review by jakebolewski — APPROVED** [review]

File: src/DataLayouts/DataLayouts.jl [hunk 1]
```diff
@@ -200,13 +200,13 @@ end
...
-    data::IJFH{S, Nij},
+    data::IJFH{S, Nij, A},
     idx::Val{Idx},
-) where {S, Nij, Idx}
+) where {S, Nij, A, Idx}
     SS = fieldtype(S, Idx)
-    T = basetype(SS)
-    offset = fieldtypeoffset(T, S, Idx)
-    nbytes = typesize(T, SS)
+    FT = eltype(A)
```
Comment (jakebolewski) [inline src/DataLayouts/DataLayouts.jl:207]: we will probably have to handle this more consistently

**Pull request merged**